### PR TITLE
GHA: build tests in a separate step from the running of them

### PR DESCRIPTION
**Pull request by bagder**
... to make the output smaller for when you want to look at test failures.

### CODE REVIEW

File: .github/workflows/msh3.yml [hunk 1]
```diff
@@ -64,10 +64,10 @@ jobs:
     - run: make V=1
       name: 'make'
 
-    - run: make V=1 examples
-      name: 'make examples'
+    - run: make V=1 -C tests
```
Comment (mback2k) [inline .github/workflows/msh3.yml:67]: Why skip building examples here?

File: .github/workflows/torture.yml [hunk 2]
```diff
@@ -54,7 +54,10 @@ jobs:
     - run: make V=1
       name: 'make'
 
-    - run: make V=1 test-torture
-      name: 'test'
+    - run: make V=1 -C tests
+      name: 'make tests'
+
+    - run: make V=1 test-ci
+      name: 'run tests'
```
Comment (mback2k) [inline .github/workflows/torture.yml:61]: Isn't the purpose of this workflow to just run torture tests?

File: .github/workflows/msh3.yml (hunk 1 above)
Comment (bagder) [reply .github/workflows/msh3.yml:67]: It was not intentional to do that change, but then again we could probably just as well skip them in this build as they are built in many others.

Comment (bagder) [reply .github/workflows/msh3.yml:67]: maybe we should consider only building the example in a single specific example job and remove the building of them from all others?

Comment (mback2k) [reply .github/workflows/msh3.yml:67]: Good idea, but we should at least cover all platforms and architectures as we had issues on some in the past with the examples.

**Pull request closed without merging**